src/classes/BDI_DataImportService.cls

```diff
@@ -337,6 +337,17 @@ global with sharing class BDI_DataImportService {
         set;
     }
 
+    /*******************************************************************************************************
+    * @description Sets diSettings based on the input settings. The method is called from tests only 
+    * since the variable is global with private set().
+    * @param diSettings Data Import Settings
+    * @return void
+    */
+    @TestVisible
+    private void injectDataImportSettings(Data_Import_Settings__c settings) {
```

Comment:
Rather than implement this new pattern for settings in tests, does it make sense to create a new method for `configDataImportSettingsForTests()` in the `UTIL_CustomSettingsFacade` class? That would match how all other custom settings are handled in unit tests.

Reply:
`diSettings` variable is only set in the `process()` method, which means test class requires an integration test just to verify that rollup triggers are disabled. Because we do not retrieve the `diSettings` from DB in the BDI class, the `UTIL_CustomSettingsFacade` usage would not work. If we inject the `diSettings`, we need *unit tests* to prove requirements by changing what settings are. The integration tests are required, but they can prove the "happy path" and rollback on error works fine. 
On the topic of `UTIL_CustomSettingsFacade`, it seems the class contains various configurations for settings. It is long, so to improve readability and extension, we might consider splitting it into specific domain custom settings (if and where required). But that is a subject for another day (or a work item :)).

Reply:
Fair enough. Settings is definitely used in a different way in this class. 
You're right about the size and complexity of the `UTIL_CustomSettingsFacade` class, but that's a story for a different story. :)